src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.cs

```diff
@@ -114,6 +114,19 @@ public static long GetTimestamp()
             return QueryPerformanceCounter();
         }
 
+        /// <summary>Gets the elapsed time since the <paramref name="startingTimestamp"/> value retrieved using <see cref="GetTimestamp"/>.</summary>
+        /// <param name="startingTimestamp">The timestamp marking the beginning of the time period.</param>
+        /// <returns>A <see cref="TimeSpan"/> for the elapsed time between the starting timestamp and the time of this call.</returns>
+        public static TimeSpan GetElapsedTime(long startingTimestamp) =>
+            GetElapsedTime(startingTimestamp, GetTimestamp());
+
+        /// <summary>Gets the elapsed time between two timestamps retrieved using <see cref="GetTimestamp"/>.</summary>
+        /// <param name="startingTimestamp">The timestamp marking the beginning of the time period.</param>
+        /// <param name="endingTimestamp">The timestamp marking the end of the time period.</param>
+        /// <returns>A <see cref="TimeSpan"/> for the elapsed time between the starting and ending timestamps.</returns>
+        public static TimeSpan GetElapsedTime(long startingTimestamp, long endingTimestamp) =>
+            new TimeSpan((long)((endingTimestamp - startingTimestamp) * s_tickFrequency));
```

Comment:
An unrelated question about `s_tickFrequency`
On Unix, we use a constant `1000000000` for frequency calculation, while on Windows we call `Kernel32.QueryPerformanceFrequency()`). Would it make sense to align them? e.g. making it constant for Windows as well, or implementing frequency shim on Unix; something like:
```c
int64_t SystemNative_GetPerformanceCounterFrequency()
{
    const int64_t defaultFrequency = 1000000000;

    struct timespec spec;
    if (clock_getres(CLOCK_MONOTONIC, &spec) != 0) return defaultFrequency;

    assert(spec.tv_sec < 1); // resolution in seconds is too high to be correct
    assert(spec.tv_nsec > 0);

    return defaultFrequency / spec.tv_nsec;
}
```

Reply:
It used to call into a native shim. It was changed in https://github.com/dotnet/runtime/pull/43343.

Reply:
For reference, Unix always returns something in terms of `seconds + nanoseconds`. Where-as Windows returns a measurement in unknown units where you have to independently query a frequency to convert it to `nanoseconds`.

The static APIs on Stopwatch are meant to be minimal overhead and directly return the "raw units" given by the underlying OS functions. This new helper is what allows it to be normalized to `TimeSpan` units, which are in terms of `100ns` ticks.